controllers/selfnoderemediationconfig_controller.go

```diff
@@ -103,6 +111,10 @@ func (r *SelfNodeRemediationConfigReconciler) syncConfigDaemonSet(ctx context.Co
 	logger := r.Log.WithName("syncConfigDaemonset")
 	logger.Info("Start to sync config daemonset")
 
+	if err := r.removeOldDsOnUpdateOperator(ctx); err != nil {
```

Comment:
What about moving this to just before the `r.syncK8sResource` call where we loop the resources we are going to sync? There we have the new DaemonSet available, and can compare its annotation value with the deployed version, and delete the deployed ds when values don't match. Then you don't need the ds name var (just take it from the new ds), and also don't need the hardcoded value var to compare with.